src/main/java/bt/dapps/SignumArt3.java

```diff
@@ -0,0 +1,531 @@
+	 *                 transfer at least this amount + contract fees)
+	 * @param timeout  how many minutes the sale will be available
+	 */
+	public void putForAuction(long priceNQT, long maxPrice, int timeout) {
+		if (highestBidder==null && owner.equals(this.getCurrentTxSender()) && unbound && bulkNFT == false) {
+			// only if there is no bidder and it is the current owner
+			status = STATUS_FOR_AUCTION;
+			auctionTimeout = getBlockTimestamp().addMinutes(timeout);
+			currentPrice = priceNQT;
+			auctionMaxPrice = maxPrice;
+			sendMessage(owner.getId(), currentPrice, auctionMaxPrice, auctionTimeout.getValue(), trackAuctionOpened);
+		}
+	}
+	
+	/**
+	 * Make an offer for an item even if that is not for sale/auction.
+	 * 
+	 * This allows for potential buyers to make an offer even for items that are not for sale.
+	 * The offer amount is locked in the contract until cancelled/accepted or a higher offer is received.
+	 * 
+	 * This offer has to be higher than a previous offer and can be cancelled later.
+	 * The owner can accept the offer and the offer is cancelled/refunded if the item is
+	 * actually sold or a running auction ends.
+	 * 
+	 */
+	public void makeOffer() {
+		if(getCurrentTxAmount() > offerPrice && highestBidder==null && unbound && bulkNFT == false )  {
+			if(offerAddress != null) {
+				// send back the latest offer
+				sendAmount(offerPrice, offerAddress);
+				sendMessage(offerAddress.getId(), offerPrice, trackOfferRemoved);
+			}
+			offerAddress = getCurrentTxSender();
+			offerPrice = getCurrentTxAmount();
+			sendMessage(offerAddress.getId(), offerPrice, trackOfferReceived);
+			
+			return;
+		}
+		// send back funds of an invalid order
+		sendAmount(getCurrentTxAmount(), getCurrentTxSender());
+	}
+	
+	/**
+	 * Cancel a previously posted offer getting back the offer amount.
+	 * 
+	 * Only the offer maker can cancel an open offer.
+	 */
+	public void cancelOffer() {
+		if(getCurrentTxSender().equals(offerAddress)) {
+			cancelOfferIfPresent();
+		}
+	}
+	
+	private void cancelOfferIfPresent() {
+		if(offerAddress == null)
+			return;
+		sendMessage(offerAddress.getId(), offerPrice, trackOfferRemoved);
+		sendAmount(offerPrice, offerAddress);
+		offerAddress = null;
+		offerPrice = ZERO;
+	}
+	
+	/**
+	 * The owner accepts a posted offer.
+	 */
+	public void acceptOffer() {
+		if(highestBidder==null && getCurrentTxSender().equals(owner) && offerPrice > ZERO && unbound && bulkNFT == false) {
+			currentPrice = offerPrice;
+			pay();
+			if (status != STATUS_NOT_FOR_SALE) {
+				sendMessage(owner.getId(), trackSetNotForSale);
+			}
+			sendMessage(offerAddress.getId(), currentPrice,owner.getId(), trackNewOwner);
+			owner = offerAddress;
+			offerAddress = null;
+			offerPrice = ZERO;
+			status = STATUS_NOT_FOR_SALE;
+			if (UseSoulbound) {
+				unbound = false;
+			}
+
+		}
+	}
+
+	// All functions needes for an BulkNFT
+	//Inital mint of the BulkSize to the owner
+	public void MintFromStack (long Quantity){
```

Comment:
methods should not start with capital letters, the same for variables: `mintFromStack`, `useSoulBound`, `mintFromStack`, `quantity`, etc.